django/src/rdwatch/tasks/__init__.py

```diff
@@ -587,3 +591,53 @@ def generate_site_images_for_evaluation_run(
             scale,
             bboxScale,
         )
+
+
+@shared_task
+def generate_image_embedding(id: int):
+    site_image = SiteImage.objects.get(pk=id)
+    if site_image:
```

Comment:
These two statements used together are redundant. Django will raise an exception if `SiteImage.objects.get(pk=id)` doesn't return a SiteImage; I _think_ there's no way for this task to be called if the `site_image` doesn't yet exist, correct? If that's true, I would say remove the conditional (which also has the advantage of decreasing the amount of indents in the code below)